temporalio/contrib/openai_agents/_temporal_model_stub.py

```diff
@@ -0,0 +1,160 @@
+        model_settings: ModelSettings,
+        tools: list[Tool],
+        output_schema: Union[AgentOutputSchemaBase, None],
+        handoffs: list[Handoff],
+        tracing: ModelTracing,
+        *,
+        previous_response_id: Union[str, None],
+        prompt: ResponsePromptParam | None,
+    ) -> ModelResponse:
+        def get_summary(input: Union[str, list[TResponseInputItem]]) -> str:
+            ### Activity summary shown in the UI
+            try:
+                max_size = 100
+                if isinstance(input, str):
+                    return input[:max_size]
+                elif isinstance(input, list):
+                    seq_input = cast(Sequence[Any], input)
+                    last_item = seq_input[-1]
+                    if isinstance(last_item, dict):
+                        return last_item.get("content", "")[:max_size]
+                    elif hasattr(last_item, "content"):
+                        return str(getattr(last_item, "content"))[:max_size]
+                    return str(last_item)[:max_size]
+                elif isinstance(input, dict):
+                    return input.get("content", "")[:max_size]
+            except Exception as e:
+                print(f"Error getting summary: {e}")
+            return ""
+
+        def make_tool_info(tool: Tool) -> ToolInput:
+            if isinstance(tool, FileSearchTool):
+                return cast(FileSearchTool, tool)
+            elif isinstance(tool, WebSearchTool):
+                return cast(WebSearchTool, tool)
+            elif isinstance(tool, ComputerTool):
+                raise NotImplementedError(
+                    "Computer search preview is not supported in Temporal model"
+                )
+            elif isinstance(tool, FunctionTool):
+                t = cast(FunctionToolInput, tool)
+                return FunctionToolInput(
+                    name=t.name,
+                    description=t.description,
+                    params_json_schema=t.params_json_schema,
+                    strict_json_schema=t.strict_json_schema,
+                )
+            else:
+                raise ValueError(f"Unknown tool type: {tool.name}")
+
+        tool_infos = [make_tool_info(x) for x in tools]
+        handoff_infos = [
+            HandoffInput(
+                tool_name=x.tool_name,
+                tool_description=x.tool_description,
+                input_json_schema=x.input_json_schema,
+                agent_name=x.agent_name,
+                strict_json_schema=x.strict_json_schema,
+            )
+            for x in handoffs
+        ]
+        if output_schema is not None and not isinstance(
+            output_schema, AgentOutputSchema
+        ):
+            raise TypeError(
+                f"Only AgentOutputSchema is supported by Temporal Model, got {type(output_schema).__name__}"
+            )
+        agent_output_schema = cast(AgentOutputSchema, output_schema)
```

Comment:
pyright types this correctly for me without a `cast`. But at any rate it should be cast to `Optional[AgentOutputSchema]` (`None` case is handled below)